net-core/Ical.Net/Ical.Net/Serialization/SimpleDeserializer.cs

```diff
@@ -0,0 +1,221 @@
+{
+    public class SimpleDeserializer
+    {
+        internal SimpleDeserializer(
+            IDataTypeMapper dataTypeMapper,
+            ISerializerFactory serializerFactory,
+            ICalendarComponentFactory componentFactory)
+        {
+            _dataTypeMapper = dataTypeMapper;
+            _serializerFactory = serializerFactory;
+            _componentFactory = componentFactory;
+        }
+
+        public static readonly SimpleDeserializer Default = new SimpleDeserializer(
+            new DataTypeMapper(),
+            new SerializerFactory(),
+            new ComponentFactory());
+
+        private const string _nameGroup = "name";
+        private const string _valueGroup = "value";
+        private const string _paramNameGroup = "paramName";
+        private const string _paramValueGroup = "paramValue";
+
+        private static Regex _contentLineRegex = new Regex(BuildContentLineRegex(), RegexOptions.Compiled);
+
+        private readonly IDataTypeMapper _dataTypeMapper;
+        private readonly ISerializerFactory _serializerFactory;
+        private readonly ICalendarComponentFactory _componentFactory;
+
+        static string BuildContentLineRegex()
+        {
+            // name          = iana-token / x-name
+            // iana-token    = 1*(ALPHA / DIGIT / "-")
+            // x-name        = "X-" [vendorid "-"] 1*(ALPHA / DIGIT / "-")
+            // vendorid      = 3*(ALPHA / DIGIT)
+            // Add underscore to match behavior of bug 2033495
+            var identifier = "[-A-Za-z0-9_]+";
+
+            // param-value   = paramtext / quoted-string
+            // paramtext     = *SAFE-CHAR
+            // quoted-string = DQUOTE *QSAFE-CHAR DQUOTE
+            // QSAFE-CHAR    = WSP / %x21 / %x23-7E / NON-US-ASCII
+            // ; Any character except CONTROL and DQUOTE
+            // SAFE-CHAR     = WSP / %x21 / %x23-2B / %x2D-39 / %x3C-7E
+            //               / NON-US-ASCII
+            // ; Any character except CONTROL, DQUOTE, ";", ":", ","
+            var paramValue = $"((?<{_paramValueGroup}>[^\\x00-\\x08\\x0A-\\x1F\\x7F\";:,]*)|\"(?<{_paramValueGroup}>[^\\x00-\\x08\\x0A-\\x1F\\x7F\"]*)\")";
+
+            // param         = param-name "=" param-value *("," param-value)
+            // param-name    = iana-token / x-name
+            var paramName = $"(?<{_paramNameGroup}>{identifier})";
+            var param = $"{paramName}={paramValue}(,{paramValue})*";
+
+            // contentline   = name *(";" param ) ":" value CRLF
+            var name = $"(?<{_nameGroup}>{identifier})";
+            // value         = *VALUE-CHAR
+            var value = $"(?<{_valueGroup}>[^\\x00-\\x19]*)";
+            var contentLine = $"^{name}(;{param})*:{value}$";
+            return contentLine;
+        }
+
+        public IEnumerable<ICalendarComponent> Deserialize(TextReader reader)
+        {
+            var context = new SerializationContext();
+            var stack = new Stack<ICalendarComponent>();
```

Comment:
Similar disclaimer as my comment on StringBuilder... I suspect stacks grow the same way `List`s do: 0, 4, 8, 16, etc. I think you're always going to have at least once round of allocation. Although, hmm... maybe the first "expansion" is free, because there's only a single allocation with no additional copy -> GC overhead. And 4 is probably a reasonable stack size to start with in this case anyway.